packages/amplify-graphql-schema-generator/src/schema-generator/schema-overrides.ts

```diff
@@ -18,16 +20,54 @@ export const applySchemaOverrides = (document: DocumentNode, existingDocument?:
         const correspondingField = findMatchingField(node, parentObjectType, existingDocument);
         if (!correspondingField) return;
 
+        // eslint-disable-next-line consistent-return
         return applyFieldOverrides(node, correspondingField);
       },
     },
   };
 
-  const updatedDocument = visit(document, schemaVisitor);
+  updatedDocument = visit(updatedDocument, schemaVisitor);
   updatedDocument['definitions'] = [...updatedDocument['definitions'], ...getNonModelTypes(existingDocument)];
+
   return updatedDocument;
 };
 
+const preserveRelationalDirectives = (document: DocumentNode, existingDocument: DocumentNode): DocumentNode => {
+  const MODEL_DIRECTIVE_NAME = 'model';
+  const RELATIONAL_DIRECTIVES = ['hasOne', 'hasMany', 'belongsTo'];
+  const documentWrapper = document as any;
+
+  if (!existingDocument) {
+    return document;
+  }
```

Comment:
This check shouldn't be required, since you don't specify as optional, and also explicitly already check this in the caller.